test/GPv2Settlement.test.ts

```diff
@@ -80,212 +75,6 @@ describe("GPv2Settlement", () => {
-
-    it("rejects transactions from non-solvers", async () => {
-      await expect(settlement.settle(...empty)).to.be.revertedWith(
-        "GPv2: not a solver",
-      );
-    });
-
-    describe("Reentrancy Protection", () => {
-      for (const { name, params } of [
-        {
-          name: "settle",
-          params: empty,
-        },
-        {
-          name: "swap",
-          params: SwapEncoder.encodeSwap(
-            [],
-            {
-              sellToken: ethers.constants.AddressZero,
-              buyToken: ethers.constants.AddressZero,
-              sellAmount: ethers.constants.Zero,
-              buyAmount: ethers.constants.Zero,
-              validTo: 0,
-              appData: 0,
-              feeAmount: ethers.constants.Zero,
-              kind: OrderKind.SELL,
-              partiallyFillable: false,
-            },
-            {
-              scheme: SigningScheme.EIP712,
-              data: `0x${"00".repeat(65)}`,
-            },
-          ),
-        },
-      ]) {
-        it(`rejects ${name} reentrancy attempts via interactions`, async () => {
-          await authenticator.connect(owner).addSolver(solver.address);
-          const encoder = new SettlementEncoder(testDomain);
-          encoder.encodeInteraction({
-            target: settlement.address,
-            callData: settlement.interface.encodeFunctionData(name, params),
-          });
-
-          await expect(
-            settlement.connect(solver).settle(...encoder.encodedSettlement({})),
-          ).to.be.revertedWith("ReentrancyGuard: reentrant call");
-        });
-
-        it(`rejects ${name} reentrancy attempt even as a registered solver`, async () => {
-          await authenticator.connect(owner).addSolver(solver.address);
-          // Add settlement contract address as registered solver
-          await authenticator.connect(owner).addSolver(settlement.address);
-          const encoder = new SettlementEncoder(testDomain);
-          encoder.encodeInteraction({
-            target: settlement.address,
-            callData: settlement.interface.encodeFunctionData(name, params),
-          });
-
-          await expect(
-            settlement.connect(solver).settle(...encoder.encodedSettlement({})),
-          ).to.be.revertedWith("ReentrancyGuard: reentrant call");
-        });
-      }
-    });
-
-    it("accepts transactions from solvers", async () => {
-      await authenticator.connect(owner).addSolver(solver.address);
-      await expect(settlement.connect(solver).settle(...empty)).to.not.be
-        .reverted;
-    });
-
-    it("emits a Settlement event", async () => {
-      await authenticator.connect(owner).addSolver(solver.address);
-      await expect(settlement.connect(solver).settle(...empty))
-        .to.emit(settlement, "Settlement")
-        .withArgs(solver.address);
-    });
-
-    it("executes interactions stages in the correct order", async () => {
```

Comment:
I'd still be in favor of keeping this test in some form, if just is a simplified fashion (just to make sure that an interaction at array position 0 is called first, at position 1 is called second, and at position 3 is called last). This could be done by having a dedicated contract that reverts if its functions aren't called in the expected order.

Reply:
This doesn't make much sense to me though as there would be nothing to stop having these interactions themselves all being in the pre-interactions, in which case the test would pass even though it hasn't executed anything in intra and post-interactions.

Reply:
This would still catch a few meaningful failure cases. For example, if `settle(..., [[0], [1], [2]])` is executed as:
```
executeInteractions(interactions[0]);
computeTradeExecutions(...);
vaultRelayer.transferFromAccounts(...);
executeInteractions(interactions[0]); // typo: index 1 is missing, index 0 is repeated
vault.transferToAccounts(...);
executeInteractions(interactions[2]);
```

Or:
```
executeInteractions(interactions[2]); // index 0 is mixed up with index 2
computeTradeExecutions(...);
vaultRelayer.transferFromAccounts(...);
executeInteractions(interactions[1]);
vault.transferToAccounts(...);
executeInteractions(interactions[0]); // index 0 is mixed up with index 2
```

It's true however that it wouldn't catch more sophisticated failure cases, but I don't think this one test needs to catch everything. The failure cases above seem more likely to me than something like:
```
executeInteractions(interactions[0]);
computeTradeExecutions(...);
vaultRelayer.transferFromAccounts(...);
executeInteractions(interactions[1]);
executeInteractions(interactions[2]); // This has been moved before `transferToAccount`
vault.transferToAccounts(...);
```

This would also be somewhat consistent with the original test: it's a bit different since it was also expected to be a way to test that `SettlementEncoder` was doing its job as expected, so two things were done at once here.

Reply:
Proposal in #178.